frontend/src/hooks/use-local-hub.tsx

```diff
@@ -1,12 +1,19 @@
 /* Copyright Contributors to the Open Cluster Management project */
 
-import { useSharedSelectors, useRecoilValue } from '../shared-recoil'
+import { useRecoilValue, useSharedAtoms } from '../shared-recoil'
 
-export function useLocalHubManagedCluster() {
-  const { localHubManagedClusterValue } = useSharedSelectors()
-  return useRecoilValue(localHubManagedClusterValue)
+export function useLocalHubName() {
+  const { localHubNameState } = useSharedAtoms()
+  const localHubName = useRecoilValue(localHubNameState)
+  if (useIsLocalCls()) return localHubName
+  else return undefined
 }
 
-export function useLocalHubName() {
-  return useLocalHubManagedCluster()?.metadata.name || 'local-cluster'
+export function useIsLocalCls() {
+  const { managedClustersState } = useSharedAtoms()
+  const managedClusters = useRecoilValue(managedClustersState)
+  const localCluster = managedClusters.find(
+    (cls) => cls.metadata?.labels && cls.metadata?.labels['local-cluster'] === 'true'
+  )
+  return !!localCluster
```

Comment:
This should not be depending on the `managedClustersState` in recoil because it depends on the user having permission to see the `ManagedCluster` resource that represents the local hub cluster. I would like to see this moved to the `/hub` backend route.